# CSE110 Assignment 2 (Loops)

This assignment is to guide in solving basic programming problems using loop in Python. 

**<font color='red'>[Please follow variable naming conventions including using meaningful variable names for all the tasks]</font>**\
<font color='red'>When you run your code, please make sure your outputs exactly match the sample outputs for each of the sample inputs given. Check if your code works for other valid inputs not given in the samples.</font>

## Write the Python code for the following problems:


In [ ]:
#Write your name, student ID and theory section number below:
#Student Name: Mohammad Sami Zaman
#Student ID: 21201207
#CSE110 Theory Section No: 03

### Task 1

Write the python programs, which prints the following sequences of values in loops:

a)	24, 18, 12, 6, 0, -6\
b)	-10, -5, 0, 5, 10, 15, 20\
c)	18, 27, 36, 45, 54, 63\
d)	18, -27, 36, -45, 54, -63

=====================================================================

**Hint(1):** You may use a while loop for solving these problems.

**Hint(2):** We are already familiar with the print() function. But, when we use it to print any value, it automatically adds an additional newline after each print statement.

For example:\
print(1)\
print(2)

Output:\
1\
2

=====================================================================

To solve this problem, in Python3, we can add an extra argument (end = " ") in the print function which tells the program to skip printing the additional newline.

For example:\
print(1, end ="")\
print(2)

Output: (prints the next output right to the previous one)\
12

=====================================================================

In Task-1(a), the loop counter may be initialized at 24 and then the loop should terminate when the loop counter reaches -6. The difference between the first two values is 24 - 18 = 6. So the loop counter value is getting decremented by 6 in every iteration for the given sequence here.

For your understanding, the code for Task 1(a) is done for you.


In [ ]:
# a) 24, 18, 12, 6, 0, -6
counter = 24
while counter >= -6:
    if counter == -6:
        print(counter, end = "") 
    else:
        print(counter, end = ", ") 
    counter = counter - 6 

24, 18, 12, 6, 0, -6

In [ ]:
# b)-10, -5, 0, 5, 10, 15, 20
counter = -10
while counter <= 20:
    if(counter <= -5):
        print(counter, end=", ")
    elif(counter < 20):
        print(counter, end=", ")
    else:
        print(counter)
    counter = counter + 5

-10, -5, 0, 5, 10, 15, 20


In [ ]:
# c)18, 27, 36, 45, 54, 63
counter = 18
while counter <= 63:
    if(counter < 63):
        print(counter, end=", ")
    else:
        print(counter)
    counter = counter + 9

18, 27, 36, 45, 54, 63


**Hints for 1(d):** 

print(5 * (-1)) gives output -5\
print("-" + str(5)) gives output -5

In [ ]:
# d)18,-27, 36,-45,54,-63
counter = 18
while counter <= 63:
    if(counter % 2==0):
        print(counter, end=", ")
    else:
        print(counter * -1, end=", ")
    counter = counter + 9

18, -27, 36, -45, 54, -63, 

### Task 2
Write a Python code for the following:

1)	Ask the user to enter the name of his favorite car.\
2)	Ask the user to enter a Number

Display the name of the user’s favorite car, the number of times specified in the second step.

==========================================================

**Example 1:** If the user enters “Toyota” and 2, your program should print the name “Toyota” two times.

**Sample Input 1:**\
Toyota\
2

**Sample Output 1:**\
Toyota\
Toyota

==========================================================

**Example 2:** If the user enters “Veyron” and 5, your program should print the name “Veyron” five times.

**Sample Input 2:**\
Veyron\
5

**Sample Output 2:**\
Veyron\
Veyron\
Veyron\
Veyron\
Veyron


In [ ]:
car = input("enter the car name=")
num = int(input("enter the number="))
count = 1
while count <= num:
    print(car)
    count = count +1

enter the car name=Toyota
enter the number=2
Toyota
Toyota


In [ ]:
car = input("enter the car name=")
num = int(input("enter the number="))
count = 1
while count <= num:
    print(car)
    count = count +1

enter the car name=Veyron
enter the number=5
Veyron
Veyron
Veyron
Veyron
Veyron


### Task 3
Write the Python code of a program that adds all numbers that are multiples of **both 7 and 9** up to 600 (including 600) i.e. 63, 126, 189, 252, ....

**The output of your program should be:** 2835\
since 63 + 126 + 189 + 252 + 315 + 378 + 441 + 504 + 567 = 2835


In [ ]:
num = 63
sum = 0
while num <= 600:
  if(num % 7==0 or num % 9==0):
    sum = sum + num
    if(num == 567):
      print(num, end=" = ")
    else:
      print(num, end=" + ")
  num = num + 63   
print(sum)

63 + 126 + 189 + 252 + 315 + 378 + 441 + 504 + 567 = 2835


### Task 4
Write the Python code of a program that adds all numbers that are multiples of **either 7 or 9** up to 600 (including 600) i.e. 7, 9, 14, 18, 21, 27, 28, 35, 36..... \
Ensure that numbers like 63, 126, 189 which are multiples of both 7 and 9 are added only once in the sum.

**The output of your program should be:** 42649


In [ ]:
num = 7
sum = 0
while num <= 600:
    if(num % 7==0 or num % 9==0):
       sum = sum + num
    num = num +1
print(sum)

42649


### Task 5
Write a Python code of a program that adds all numbers that are multiples of **either 7 or 9 but not both**, up to 600 (including 600) i.e. 7, 9, 14, 18, 21..... and so on but not the numbers 63, 126, 189..... which are multiples of both 7 and 9.

**The output of your program should be:** 39814


In [ ]:
num = 7
sum = 0
while num <= 600:
    if(num % 7==0 and num % 9==0):
       sum = sum
    elif(num % 7==0 or num % 9==0):
        sum = sum + num
    num = num +1
print(sum)

39814


### Task 6
Write the Python code of a program that displays all the **odd numbers** between 10 and 50 (inclusive).

**Output:** 11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49


In [ ]:
num = 11
while num <= 50:
    print(num, end=" ")
    num = num + 2

11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49 

### Task 7
Write a Python code that will calculate the **value of y if the expression** of y is as follows (n is the input):  

$y = 1^2 - 2^2 +3^2 -4^2 +5^2 .........+ n^2 $

==========================================================

**Sample Input 1:**\
5

**Sample Output 1:**\
15

**Explanation:** y = 1 - 4 + 9 - 16 + 25 = 15

==========================================================

**Sample Input 2:**\
10

**Sample Output 2:**\
-55

**Explanation:** y = 1 - 4 + 9 - 16 + 25 - 36 + 49 - 64 + 81 - 100 = -55

==========================================================

**Sample Input 3:**\
20

**Sample Output 3:**\
-210



In [ ]:
var_n = int(input('enter the number='))
var_y = 0
count = 1
while count <= var_n:
    if(count == var_n):
        print(count**2, end=' = ')
    elif(count % 2==0):
        print(count**2, end=' - ')
    else:
        print(count**2, end=' + ')
    if(count % 2==0):
        var_y = var_y - count**2
    else:
        var_y = var_y + count**2
    count = count + 1
print(var_y)

enter the number=5
1 + 4 - 9 + 16 - 25 = 15


In [ ]:
var_n = int(input('enter the number='))
var_y = 0
count = 1
while count <= var_n:
    if(count == var_n):
        print(count**2, end=' = ')
    elif(count % 2==0):
        print(count**2, end=' - ')
    else:
        print(count**2, end=' + ')
    if(count % 2==0):
        var_y = var_y - count**2
    else:
        var_y = var_y + count**2
    count = count + 1
print(var_y)

enter the number=10
1 + 4 - 9 + 16 - 25 + 36 - 49 + 64 - 81 + 100 = -55


In [ ]:
var_n = int(input('enter the number='))
var_y = 0
count = 1
while count <= var_n:
    if(count == var_n):
        print(count**2, end=' = ')
    elif(count % 2==0):
        print(count**2, end=' - ')
    else:
        print(count**2, end=' + ')
    if(count % 2==0):
        var_y = var_y - count**2
    else:
        var_y = var_y + count**2
    count = count + 1
print(var_y)

enter the number=20
1 + 4 - 9 + 16 - 25 + 36 - 49 + 64 - 81 + 100 - 121 + 144 - 169 + 196 - 225 + 256 - 289 + 324 - 361 + 400 = -210


### Task 8

Write a Python code of a program that asks the user to enter ten numbers and then display the total and the average of **ONLY** the **odd numbers** among those ten numbers.\
<font color='red'>[Please do not use list for this task]</font>

==========================================================


**Sample Input 1:**\
1

2

3

4

5

6

7

8

9

10

**Sample Output 1:**
The total of the odd numbers is 25 and their average is 5.0


**Explanation:** The total is 1 + 3 + 5 + 7 + 9 = 25 and the average is 25/5 = 5.0

==========================================================


**Sample Input 2:**\
-20

13

-5

40

-17

10

20

-8

99
  
-200

**Sample Output 2:**

The total of the odd numbers is 89 and their average is 17.8


**Explanation:** The total is 90 = 13 + (-5) + (-17) + 99) and their average is 90/4 = 22.5

In [ ]:
count = 1
total = 0
while count <= 10:
   var_x = float(input('please enter the number ='))
   if(var_x % 2 == 0):
      total = total
   else:
      total = total + var_x
   avg = total / 5
   count = count + 1
print("the total of odd number is", total)
print('the avarage of odd number is', avg)

please enter the number =1
please enter the number =2
please enter the number =3
please enter the number =4
please enter the number =5
please enter the number =6
please enter the number =7
please enter the number =8
please enter the number =9
please enter the number =10
the total of odd number is 25.0
the avarage of odd number is 5.0


In [ ]:
count = 1
total = 0
while count <= 10:
   var_x = float(input('please enter the number ='))
   if(var_x % 2 == 0):
      total = total
   else:
      total = total + var_x
   avg = total / 5.0
   count = count + 1
print("the total of odd number is", total)
print('the avarage of odd number is', avg)

please enter the number =-20
please enter the number =13
please enter the number =-5
please enter the number =40
please enter the number =-17
please enter the number =10
please enter the number =20
please enter the number =-8
please enter the number =99
please enter the number =-200
the total of odd number is 90.0
the avarage of odd number is 18.0


### Task 9
Write a Python code for the following:
* Ask the user to enter a Number, N
* Display the summation of multiples of 7 up to that number (**from 1 to N inclusive**)

========================================================== 

**Sample Input 1:**\
50

**Sample Output 1:**\
196

**Explanation:** The summation of multiples of 7 up to 50 is 7 + 14 + 21 + 28 + 35 + 42 + 49 = 196

==========================================================


**Sample Input 2:**\
75

**Sample Output 2:**\
385

**Explanation:** The summation of multiples of 7 up to 75 is 7 + 14 + 21 + 28 + 35 + 42 + 49 + 56 + 63 + 70 = 385





In [ ]:
var_n = int(input('enter the number'))
count = 1
sum = 0
while count <= var_n:
    if(count % 7==0):
        sum = sum + count
    count = count + 1
print(sum)

enter the number50
196


In [ ]:
var_n = int(input('enter the number'))
count = 1
sum = 0
while count <= var_n:
    if(count % 7==0):
        sum = sum + count
    count = count + 1
print(sum)

enter the number75
385


### Task 10
Write a Python code that will read 5 numbers from the user. Your program should print the first number, the sum of the first 2 numbers, the sum of the first 3 numbers, and so on up to the sum of 5 numbers.

==========================================================

**Sample Input 1:**\
1\
2\
3\
4\
5

**Sample Output 1:**\
1\
3\
6\
10\
15

**Explanation:**\
When the user enters 1 at first, we print 1, then \
The user enters 2, (1 + 2) = 3, so our output is 3, then \
The user enters 3, (3 + 3) = 6, hence our output is 6, then \
The user enters 4, and we have (6 + 4) = 10, we print 10, and finally\
The user enters 5, the final sum is 15 which is printed as the last output.

==========================================================

**Sample Input 2:**\
11\
-29\
-3\
47\
50

**Sample Output 2:**\
11\
-18\
-21\
26\
76

**Explanation:**\
When the user enters 11 at first, we print 11, then \
The user enters -29, (11 + (-29)) = -18, so our output is -18, then \
The user enters -3, (-18 - 3) = -21, hence our output is -21, then \
The user enters 4, and we have (-21 + 47) = 26, we print 26, and finally\
The user enters 50, the final sum is (26 + 50) = 76 which is printed as the last output.


In [ ]:
count = 1
sum = 0
while count <= 5:
    var_n = int(input('the number = '))
    sum = sum + var_n
    print(sum)
    count = count + 1

the number = 1
1
the number = 2
3
the number = 3
6
the number = 4
10
the number = 5
15


In [ ]:
count = 1
sum = 0
while count <= 5:
    var_n = int(input('the number = '))
    sum = sum + var_n
    print(sum)
    count = count + 1

the number = 11
11
the number = -29
-18
the number = -3
-21
the number = 47
26
the number = 50
76





### Task 11
Write a Python program which takes a number and prints the digits from the unit place, then the tenth, then hundredth, etc. <font color='red'>(Right to Left)</font>

<font color='red'>[Consider the input number to be an INTEGER. You are not allowed to use String indexing for solving this task]</font>

**Example:** If the user gives 32768, then print 8, 6, 7, 2, 3 

=========================================================================

**Hint (1):**  The input() function, converts the input data to String data type by default. Therefore, please type cast it to an integer before proceeding further.


**Hint  (2):** First to get the digit from the right side, we can take the remainder of the number using modulus (%) operator i.e. mod 10 to get the rightmost digit and print it. For dropping the last digit, we can perform floor division by 10 on the number and then continue the same to print the other digits as shown below.

32768 % 10 = 8 

32768 // 10 = 3276 

Then,

3276 % 10 = 6

3276 // 10 = 327 

and so on

327 % 10 = 7 

327 // 10 = 32 

32 % 10 = 2 

32 // 10 = 3 

3 % 10 = 3 

3 // 10 = 0 

Done! When the number becomes 0 you can end your loop.



In [ ]:
var_n = int(input('enter the number = '))
while var_n // 2 != 0:
    var_m = var_n % 10
    if var_m % var_n == 0:
        print(var_m)
    else:
        print(var_m, end=', ')
    var_n = var_n // 10

enter the number = 32768
8, 6, 7, 2, 3


### Task 12

Write a Python program that takes a number and prints how many digits are in that number. \
<font color='red'>[Consider the input number to be an INTEGER.]</font>

<font color='red'>[You are not allowed to use len() function]</font>

**Example:** If the user gives 9876, your program should print 4. 



**Hint:** We may keep floor dividing by ten and count how many times this can be done until the number becomes 0 as illustrated below. 

9876 // 10 = 987, count = 1 \
then, 987 // 10 = 98, count = 2 \
98 // 10 = 9, count = 3 \
9 // 10 = 0, count = 4 

Done! When the number becomes 0 your loop can end giving you the count of digits, in this case for our input of 9876, 4 digits.



In [ ]:
var_n = int(input('enter the number = '))
count = 0
while var_n // 2 != 0:
    count = count +1
    var_n = var_n // 10
print('the number digits are=', count)

enter the number = 98765
the number digits are= 5


### Task 13
Write a Python program that takes a number from the user and prints its digits from  <font color='red'> left to right. </font>

<font color='red'>[Consider the input number to be an INTEGER. You are not allowed to use String indexing for solving this task]</font>

=========================================================================

**Example:** if the user gives 32768, then print 3, 2, 7, 6, 8 

=========================================================================

**Hint(1):**  The input() function takes the input data as String data type by default. Please convert it to an integer before starting your code for the task.

**Hint(2):**  

Step 1: First, count how many digits\
Step 2: Then, calculate 10 to the power (number of digits - 1).

Step 3 with explanation: Say, the input given by the user as in our example, 32768 has 5 digits, so calculating 10 to the power 4 gives us 10000. Then floor dividing 32768 by 10000, we can get 3.\
Proceeding further, the remainder of 32768 by 10000 (32768 % 10000) gives us 2768. This time to get 2 we need to floor divide 2768 by 1000 which is basically our 10000/10. Again, taking remainder of 2768 by 1000 gives us 768 which we then divide by 100 (i.e. 1000/10) and keep on doing this until there are no more digits left (zero).


**To summarize and clarify:**\
Loop 1: First, we count digits like our Task 12, say 5 in this case for 32768\
Loop 2: Then, we calculate 10 to the power 4 (5-1), that is 10000.\
Loop 3: Then we keep repeating the three steps of floor dividing, modulus and dividing by 10 as demonstrated below.

32768 // 10000 = 3\
32768 % 10000 = 2768 \
10000 // 10 = 1000

2768 // 1000 = 2\
2768 % 1000 = 768\
1000 // 10 = 100

768 // 100 = 7\
768 % 100 = 68 \
100 // 10 = 10

68 // 10 = 6\
68 % 10 = 8 \
10 // 10 = 1

8 // 1 = 8\
8 % 1 = 0 \
1 // 10 = 0

Done. Loop ends as number has become 0.

In [ ]:
var_n = int(input('enter the number= '))
count = 0
while var_n // 2 != 0:
    count = count +1
    var_n = var_n // 10
print('the number digits are=', count)
var_m = int(input('enter the same number= '))
count = count
var_x = (10 ** (count - 1))
while var_x != 0:
    var_a = var_m // var_x
    if var_m % var_x == 0:
        print(var_a)
    else:
        print(var_a, end=', ')
    var_m = var_m % var_x
    var_x = var_x // 10

enter the number= 32768
the number digits are= 5
enter the same number= 32768
3, 2, 7, 6, 8


### Task 14

Write a Python program that takes a number as input from the user and prints the divisors of that number as well as how many divisors the number has.

==========================================================

**Sample Input 1:**\
6

**Sample Output 1:**\
1, 2, 3, 6 \
Total 4 divisors.


==========================================================

**Sample Input 2:**\
121

**Sample Output 2:**\
1, 11, 121 \
Total 3 divisors.




In [ ]:
var_n = int(input('enter the number= '))
count = 1
while count <= var_n:
    if var_n % count == 0:
        if count == var_n:
            print(count)
        else:
            print(count, end=', ')
    count = count + 1

enter the number= 6
1, 2, 3, 6


In [ ]:
var_n = int(input('enter the number= '))
count = 1
while count <= var_n:
    if var_n % count == 0:
        if count == var_n:
            print(count)
        else:
            print(count, end=', ')
    count = count + 1

enter the number= 121
1, 11, 121


### Task 15
Write a Python program that takes a number as input from the user and tells if it is a perfect number or not.

**Perfect Number:** An integer number is said to be a perfect number if its factors, including 1 but not the number itself, sum to the number.

==========================================================

**Sample Input 1:**\
6

**Sample Output 2:**\
6 is a perfect number

**Explanation:**\
6 has 4 divisors: 1, 2, 3, and 6. \
If we add all factors except 6 itself, 1 + 2 + 3 = 6. The sum of the factors excluding the number itself sum up to the number therefore "6 is a perfect number" is printed.

==========================================================

**Sample Input 2:**\
28

**Sample Output 2:**\
28 is a perfect number

**Explanation:**\
28 has the divisors: 1, 2, 4, 7, 14, and 28. \
If we add all factors except itself we get 28, 1 + 2 + 4 + 7 + 14 = 28.

==========================================================

**Sample Input 3:**\
33

**Sample Output 3:**\
33 is not a perfect number

**Explanation:**\
33 has 4 divisors: 1, 3, 11, and 33. \
If we add all factors except itself, 15 = 1 + 3 + 11. The sum is not equal to the number, therefore 33 is not a perfect number.



In [ ]:
var_n = int(input('enter the number= '))
count = 1
sum = 0
while count <= var_n:
    if var_n % count == 0:
        if count == var_n:
            print(count)
            
        else:
            print(count, end=', ')
            sum = sum +count    
    count = count + 1
if sum == var_n:
        print(var_n, 'is perfect number')
else:
        print(var_n, 'is not perfect number')  

enter the number= 28
1, 2, 4, 7, 14, 28
28 is perfect number


In [ ]:
var_n = int(input('enter the number= '))
count = 1
sum = 0
while count <= var_n:
    if var_n % count == 0:
        if count == var_n:
            print(count)
            
        else:
            print(count, end=', ')
            sum = sum +count    
    count = count + 1
if sum == var_n:
        print(var_n, 'is perfect number')
else:
        print(var_n, 'is not perfect number')  

enter the number= 33
1, 3, 11, 33
33 is not perfect number


### Task 16

Write a Python program that asks the user for one number and tells if it is a prime number or not.

**Prime Number:** If a number has only two divisors, (1 and itself), then it is a prime number. If it is divisible by more numbers, then it is not a prime.

**Hint:** You may use the code for counting divisors from Task 14. 

==========================================================

**Sample Input 1:**\
11

**Sample Output 1:**\
11 is a prime number

**Explanation:** 11 has only 2 divisors: 1 and 11.

==========================================================

**Sample Input 2:**\
6

**Sample Output 2:**\
6 is not a prime number

**Explanation:** 6 has 4 divisors: 1, 2, 3 and 6.






In [ ]:
var_n = int(input('enter the number = '))
if var_n > 1:
    for count in range(2, var_n // 2):
        if(var_n % count == 0):
            print(var_n, 'is not a prime number')
            break
        else:
            print(var_n, 'is a prime number')
            break
else:
    print(var_n, 'is not a prime number')  

enter the number = 11
11 is a prime number


In [ ]:
var_n = int(input('enter the number = '))
if var_n > 1:
    for count in range(2, var_n // 2):
        if(var_n % count == 0):
            print(var_n, 'is not a prime number')
            break
        else:
            print(var_n, 'is a prime number')
            break
else:
    print(var_n, 'is not a prime number')  

enter the number = 6
6 is not a prime number


### Task 17

Write a Python program that asks the user for a quantity, then takes that many numbers as input and prints the maximum, minimum and average of those numbers. 

<font color='red'>[Please note that you CANNOT use max, min built-in functions]

[Also, you DO NOT need to use lists for this task]</font>

==========================================================

**Example:** If the user enters 5 as an input for quantity and the enters the 5 numbers, 10, 4, -1, -100, and 1. \
The output of your program should be: “Maximum 10”, “Minimum -100”, “Average is -17.2” as shown below.

**Input:**\
5
10
4
-1
-100
1

**Output:**\
Maximum 10\
Minimum -100\
Average is -17.2

**Explanation:** Average calculation: (10 + 4 + (-1) + (-100) + 1)/5 = -86/5 = -17.2



In [9]:
count = 1
var_s = 0
max_val = None
min_val = None
while count <= 5:
    var_n = int(input('enter the number = '))
    var_s = var_s + var_n
    if count == 1:
        max_val = var_n
        min_val = var_n
    else:
        if var_n > max_val:
            max_val = var_n
        if var_n < min_val:
            min_val = var_n        
    count = count +1
print('max value', max_val)
print('min value', min_val)
avg = var_s / 5
print('average is', avg)

enter the number = 10
enter the number = 4
enter the number = -1
enter the number = -100
enter the number = 1
max value 10
min value -100
average is -17.2


## Task 18

Write a python program that prints a square of size N using + where N will be given as input as illustrated in the examples below.

**Hint:** You may need to use nested loops to solve this problem.

==========================================================

**Sample Input 1:**

5

**Sample Output 1:**<br>
+++++<br>
+++++<br>
+++++<br>
+++++<br>
+++++<br>

**Explanation:** Here, we may see it as 5 rows and 5 columns, where in each horizontal row/line, 5 '+' is printed next to each other and we have a total of 5 lines.

==========================================================

**Sample Input 2:**

3

**Sample Output 2:**<br>
+++<br>
+++<br>
+++<br>

**Explanation:** If the input is 3, then there will be 3 rows and 3 columns, where in each horizontal row/line, '+' is printed 3 times, 1 in each of the 3 columns and we will have a total of 3 lines.

In [ ]:
col_num = int(input('enter the number = '))
for x in range(1, col_num + 1):
    for y in range(1, col_num + 1):
        if y <= x:
            print('+', end='')
        else:
            print('+', end='')
    print('')

enter the number = 5
+++++
+++++
+++++
+++++
+++++


In [ ]:
col_num = int(input('enter the number = '))
for x in range(1, col_num + 1):
    for y in range(1, col_num + 1):
        if y <= x:
            print('+', end='')
        else:
            print('+', end='')
    print('')

enter the number = 3
+++
+++
+++


### Task 19

Write a python program that prints a rectangle of size M (height/line numbers) and N (length/column numbers) using incrementing numbers where M and N will be given as input. Please look at the examples below for clarification.

**Hint:** You may need to use nested loops and print the loop counter variable in one of the loops.

==========================================================

**Sample Input 1:**\
4\
6

**Sample Output 1:**<br>
123456<br>
123456<br>
123456<br>
123456<br>

**Explanation:** The user has given 4 rows/lines and 6 columns as input. Therefore, we have 4 lines in our output here and in each line, the column number is printed from 1 through to 6 for the 6 columns.

==========================================================

**Sample Input 2:**\
3\
2

**Sample Output 2:**<br>
12<br>
12<br>
12<br>

**Explanation:** Our user input this time is 3 rows/lines and 2 columns. So, our output has 3 lines and in each line, the column number 1 and 2 are printed next to each other as the user only wants 2 columns here.


In [ ]:
row_n = int(input('enter the row number = '))
col_m = int(input('enter the coloum number = '))
for x in range(1, row_n + 1):
    for y in range(1, col_m + 1, 1):
          print(y, end='')
    print('')

enter the row number = 3
enter the coloum number = 2
12
12
12


In [ ]:
row_n = int(input('enter the row number = '))
col_m = int(input('enter the coloum number = '))
for x in range(1, row_n + 1):
    for y in range(1, col_m + 1, 1):
          print(y, end='')
    print('')

enter the row number = 4
enter the coloum number = 6
123456
123456
123456
123456


### Task 20

Write a python program that prints a right angled triangle of height N using incrementing numbers where N will be given as input.

**Hint:** You may need to use nested loops. Try to think up to which point should the inner loop run.

==========================================================

**Sample Input 1:**\
4

**Sample Output 1:**<br>
1<br>
12<br>
123<br>
1234<br>

**Explanation:** For an input of 4, we have 4 rows/lines where in each line, the respective column number is printed sequentially up to the line/row number. So, in line number 1, we have 1 only. In line 2, 12 is printed. In line 3, we have 123 and so on.

==========================================================

**Sample Input 2:**\
5

**Sample Output 2:**<br>
1<br>
12<br>
123<br>
1234<br>
12345<br>

**Explanation:** Numbers are printed sequentially up to the line number for each of the lines.

In [ ]:
var_n = int(input('enter the number = '))
for x in range(1, var_n + 1):
    for y in range(1, x + 1):
        print(y, end='')
    print('')

enter the number = 5
1
12
123
1234
12345


In [ ]:
var_n = int(input('enter the number = '))
for x in range(1, var_n + 1):
    for y in range(1, x + 1):
        print(y, end='')
    print('')

enter the number = 4
1
12
123
1234


## Optional Tasks (21 -25) [Ungraded]

**These tasks are just for practice. No marks will be deducted for not completing them and no extra marks will be given for solving them. Just try and practice these advanced problems.**

### Task 21

Write a python program that prints all the fibonacci numbers from 0 to N where N will be given as input by the user. 

**A Fibonacci number is a number which is the summation of its previous two fibonacci numbers.**

The first two fibonacci numbers are 0 and 1. 

So, the 3rd is 0 + 1 = 1, 

the 4th is 1 + 1 = 2, 

the 5th is 1 + 2 = 3,

the 6th one is 2 + 3 = 5,

and so on. 

i.e. 0, 1, 1, 2, 3, 5, 8, 13, 21.......

==========================================================

**Sample Input 1:**<br>
10

**Sample Output 1:**<br>
0 1 1 2 3 5 8

**Explanation**: All fibonacci numbers up to the number 10 is printed separated by spaces.

==========================================================

**Sample Input 2:**<br>
15

**Sample Output 2:**<br>
0 1 1 2 3 5 8 13

**Explanation**: All fibonacci numbers up to the number 15 is printed separated by spaces.

In [ ]:
#Todo

### Task 22

Write a python program that converts a decimal integer number to a binary number. 

*Hint: A decimal can be converted to a binary number by keeping track of the remainders after each division of that number by 2.*

==========================================================

**Example 1:**

For example, to convert 11 to a binary number, we can try the following approach.

11 // 2 = 5 (Remainder 1) 
<br>5 // 2 = 2 (Remainder 1) 
<br>2 // 2 = 1 (Remainder 0) 
<br>1 // 2 = 0 (Remainder 1) 

Now, taking the remainders from bottom to top gives us the binary number which is 1011. Binary of 11 is 1011.

**Sample Input 1:** <br>
11

**Sample Output 1:**<br>
1011

==========================================================

**Example 2:**

Similarly, to convert 13 to a binary number:

13 // 2 = 6 (Remainder 1) 
<br>6 // 2 = 3 (Remainder 0) 
<br>3 // 2 = 1 (Remainder 1) 
<br>1 // 2 = 0 (Remainder 1) 

Again, taking the remainders from the bottom to top, which is 1101 gives us the binary number. Binary of 13 is 1101.

**Sample Input 2:** <br>
13

**Sample Output 2:**<br>
1101

In [ ]:
#Todo

### Task 23

Write a python program that converts a binary number to a decimal integer number after taking the binary number as an input from the user. 

**Note: A binary number can be converted to its corresponding decimal number by multiplying each binary digit with a power of 2, where the power denotes the position of the binary digit**

==========================================================

**Example 1:**

For example, to convert 1010 to decimal, we can do this


$2^{3}$ x 1 + $2^{2}$ x 0 + $2^{1}$ x 1 + $2^{0}$ x 0 = 10


**Sample Input 1:** <br>
1010

**Sample Output 1:** <br>
10

==========================================================

**Example 2:**

So, to convert 11101, we can do this


$2^{4}$ x 1 + $2^{3}$ x 1 + $2^{2}$ x 1 + $2^{1}$ x 0 + $2^{0}$ x 1 = 29


**Sample Input 2:** <br>
11101

**Sample Output 2:** <br>
29

In [ ]:
#Todo

### Task 24
Write a Python program that asks the user for a range (a starting number and an ending number) and then 
**counts** how many numbers are prime numbers and how many numbers are perfect numbers between that range. It also **prints** those numbers in the format shown in the output of the example given below. 


=========================================================================

**Hint (1):** We may declare two strings/lists to store the prime and perfect numbers. Inside the loop, we check for prime and perfect numbers and add them to respective string/list besides counting them.

**Hint (2):** We may need to convert numbers in String to integers for checking and calculations. We may again need to convert numbers to Strings for storing and printing as the outputs shown in the example below.

=========================================================================

**Example:** If the user gives us a range of between 2 and 6, there are 3 prime numbers (2, 3, 5) and 1 perfect number (6) in that range. 

**Input:**\
2\
6

**Output:**\
Between 2 and 6,\
Found 3 prime numbers \
Found 1 perfect number\
Prime numbers: 2,  3,  5 \
Perfect number: 6 



In [ ]:
#Todo

###Task 25

Write a python program that asks the user for a range, a starting number(inclusive) and an ending number (inclusive). Then, take another input to for checking. If the product of all the digits of each number in the range is divisible by the third input, then print the number.

=========================================================================

**Sample Input 1:**\
25\
30\
4 

**Sample Output 1:**\
12 16 0

**Explanation:** The product of all the digits of each number from 25 to 30 are 2 x 5, 2 x 6, 2 x 7, 2 x 8, 2 x 9 and 3 x 0. The final products are 10, 12, 14, 16, 18 and 0 respectively. Out of these numbers only 12, 16 and 0 are divisible by the third input 4 and therefore they are printed.

=========================================================================

**Sample Input 2:**\
351\
356\
9

**Sample Output 2:**\
45 90

**Explanation:** The product of all the digits of each number from 351 to 356 are 3x5x1, 3x5x2, 3x5x3, 3x5x4, 3x5x5 and 3x5x6. The final products are 15, 30, 45, 60, 75 and 90 respectively. Out of these numbers only 45 and 90 are divisible by 9 and therefore they are printed.

=========================================================================
